lib/gateway/socket-connection.js

```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+const WebSocket = require('ws')
+
+function createClient (url, subscriber, serviceName) {
+  const ws = new WebSocket(url, 'graphql-ws')
+  const client = WebSocket.createWebSocketStream(ws, { encoding: 'utf8', objectMode: true })
+  client.setEncoding('utf8')
+  client.write(JSON.stringify({
+    type: 'connection_init'
+  }))
+
+  client.on('data', async (chunk) => {
```

Comment:
Don't do this. This is going to leak memory in case of errors.

Reply:
This was replaced by the `subscription-client.js`